src/main/java/com/xpn/xwiki/plugin/lucene/LucenePlugin.java

```diff
@@ -142,7 +137,7 @@
    */
   private String indexDirs;
 
-  private IndexRebuilder indexRebuilder;
+  IndexRebuilder indexRebuilder;
```

Comment:
indexRebuilder must be volatile or all accesses (flushCache and getters) must be synchronized.